### PR TITLE
Forms: add post to url hook

### PR DESCRIPTION
**Pull request by CGastrell**
This PR implements the post-to-url feature on the Forms package as a replacement to the (left behind) [Salesforce integration implementation on the plugin](https://github.com/Automattic/jetpack/pull/26903).

## Proposed changes:
Add a new class to hook on feedback post creation. If the form attributes specify a post-to-url, it will trigger an HTTP post using the submit values. Salesforce configurations are accepted and turned into a generic configuration as a backwards compatibility feature.

The change also adds new expected attributes on the form to pave the road for new features:
- hiddenFields: TBD, simple name/value pairs added to forms
- postToUrl: TBD, post to url configuration
- salesforceData (backwards compatibility with plugin)

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

Make sure you enable the package:
```php
add_filter( 'jetpack_contact_form_use_package', '__return_true' );
```

Run install on both the plugin and package (jetpack install packages/forms && jetpack install plugins/jetpack) to ensure autoloader update.

Publish a post with a Salesforce Lead form (or use an existing one). Make sure you use a valid `organizationID` on the form setup. Fill the form and submit. You should see a new lead created on your Salesforce instance.

Salesforce instance can be acquired free for testing, I can't share mine. Use the form's instructions to get your organization ID.

### CODE REVIEW

File: projects/packages/forms/src/class-jetpack-forms.php [hunk 1]
```diff
@@ -62,4 +62,14 @@ public static function is_feedback_dashboard_enabled() {
...
 	}
+
+	/**
+	 * Initialization, calls for register pattern and loads any extra file/libs.
+	 */
+	public static function init() {
+		Util::register_pattern();
+
+		// Load any extra file/libs so they are available on the namespace for use (Forms\Post_To_URL_Hook::initialize)
+		require_once __DIR__ . '/class-post-to-url-hook.php';
```
Comment (ice9js) [inline projects/packages/forms/src/class-jetpack-forms.php]: I don't understand why we need this at all? The file should already be loaded through the autoloader and you can just proceed to use `Automattic\Jetpack\Forms\Post_To_Url_Hook` where necessary.

File: projects/packages/forms/src/class-post-to-url-hook.php [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+ */
+
+namespace Automattic\Jetpack\Forms;
+
+/**
+ * Class Post_To_Url_Hook
+ *
+ * Hooks on Jetpack's Contact form to post form data to some URL.
+ */
+class Post_To_Url_Hook {
```
Comment (ice9js) [inline projects/packages/forms/src/class-post-to-url-hook.php]: Not sure about the name of this class quite yet - it seems the Salesforce connection is currently hardcoded to let's name it accordingly.

Either way I think it'd make more sense inside the `service/` subdirectory - where the google drive class is located as well. Make sure to update the namespace as well when you move it.

**Review by ice9js — COMMENTED** [review]
I left some initial comments on code organization, still need to test this.

File: projects/packages/forms/src/class-post-to-url-hook.php (hunk 2 above)
Comment (CGastrell) [reply projects/packages/forms/src/class-post-to-url-hook.php]: > Not sure about the name of this class quite yet - it seems the Salesforce connection is currently hardcoded to let's name it accordingly.

Well, actually that's the big change here: it should be able to handle any form with the proper configuration to send submissions to a remote URL. Salesforce is just one configuration and here is being adapted to this new, more generic syntax/structure. Agree on the naming though.

> Either way I think it'd make more sense inside the service/ subdirectory - where the google drive class is located as well. Make sure to update the namespace as well when you move it.

Makes sense.

File: projects/packages/forms/src/class-jetpack-forms.php (hunk 1 above)
Comment (CGastrell) [reply projects/packages/forms/src/class-jetpack-forms.php]: Right! I was installing the package with no success, then I realized I needed to install the plugin for the autoloader to update. Thanks!

**Review by ice9js — DISMISSED** [review]
Looks good, thanks Christian!

**Review by ice9js — APPROVED** [review]

**Pull request merged**